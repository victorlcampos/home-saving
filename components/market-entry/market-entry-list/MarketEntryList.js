import React from 'react';
import { Text,
         FlatList,
         StyleSheet,
         View,
         Button,
         Platform,
         TouchableOpacity } from 'react-native';
import { Ionicons } from '@expo/vector-icons';

export class MarketEntryList extends React.Component {
  extractKey = ({id}) => id

  iconName = ({checked}) => {
    return Platform.OS === 'ios' ?
                      `ios-checkmark-circle${checked ? '' : '-outline'}` :
                      `md-checkmark-circle${checked ? '' : '-outline'}`;
  }

  renderItem = ({item}) => {
    return (
      <View style={styles.row} >
        <Text style={{fontSize: 24}}>
          {item.name}
        </Text>

        <TouchableOpacity onPress={this.props.checkItem(item)} >
          <View style={styles.button}>
            <Ionicons name={this.iconName(item)} size={46} />
          </View>
        </TouchableOpacity>
      </View>
    )
  }

  render() {
    const {entries} = this.props;
    this.state = entries;

    return (
      <FlatList
        style={styles.container}
        data={entries}
        renderItem={this.renderItem}
        keyExtractor={this.extractKey}
      />
    );
  }
}

const styles = StyleSheet.create({
  container: {
    marginTop: 20,
    flex: 1
  },
  row: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-between',
    padding: 15,
    marginBottom: 5,
    backgroundColor: '#FFF',
    borderBottomWidth: 1,
    borderBottomColor: '#EEE',
    alignItems: 'center',
  },
  button: {
    padding: 5
  }
})
