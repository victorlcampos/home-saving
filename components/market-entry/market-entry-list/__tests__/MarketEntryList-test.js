import { Text } from 'react-native';
import React from 'react';
import { MarketEntryList } from '../MarketEntryList';
import renderer from 'react-test-renderer';

it('render all entries', async () => {
  const state = {
    entries: [
      {id: 0, name: 'Frutas'},
      {id: 1, name: 'Verduras'},
    ]
  }

  handlers = {
    checkItem: (item) => {
      return () => {
      }
    }
  }

  const tree =  renderer.create(<MarketEntryList {...state} {...handlers} />).toJSON();
  expect(tree).toMatchSnapshot();
});
