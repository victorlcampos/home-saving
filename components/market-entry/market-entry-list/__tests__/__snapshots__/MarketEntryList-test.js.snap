// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render all entries 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "id": 0,
        "name": "Frutas",
      },
      Object {
        "id": 1,
        "name": "Verduras",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "flex": 1,
      "marginTop": 20,
    }
  }
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#CFCFCF",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 5,
            "padding": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 24,
            }
          }
        >
          Frutas
        </Text>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "padding": 5,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#CFCFCF",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 5,
            "padding": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 24,
            }
          }
        >
          Verduras
        </Text>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "padding": 5,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
